blocks/library/categories/data.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns a Promise with the categories or an error on failure.
+ *
+ * @returns {wp.api.collections.Categories} Returns a Promise with all categories.
+ */
+export function getCategories() {
+	const categoriesCollection = new wp.api.collections.Categories();
+
+	const categories = categoriesCollection.fetch();
```

Comment:
Will this not introduce the possibility of there being _many_ requests when there is a large deeply-nested category tree?